packages/frontend/src/features/sqlRunner/hooks/useSqlQueryRun.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+    SchedulerJobStatus,
+    type ApiError,
+    type ApiJobScheduledResponse,
+    type ApiJobStatusResponse,
+    type ResultRow,
+    type SqlRunnerBody,
+} from '@lightdash/common';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { lightdashApi } from '../../../api';
+import useToaster from '../../../hooks/toaster/useToaster';
+import { getSchedulerJobStatus } from '../../scheduler/hooks/useScheduler';
+import { useAppSelector } from '../store/hooks';
+
+const scheduleSqlJob = async ({
+    projectUuid,
+    sql,
+}: {
+    projectUuid: string;
+    sql: SqlRunnerBody['sql'];
+}) =>
+    lightdashApi<ApiJobScheduledResponse['results']>({
+        url: `/projects/${projectUuid}/sqlRunner/run`,
+        method: 'POST',
+        body: JSON.stringify({ sql }),
+    });
+
+/**
+ * Gets the SQL query results from the server
+ *
+ * Steps:
+ * 1. Schedule the SQL query job
+ * 2. Get the status of the scheduled job
+ * 3. Fetch the results of the job
+ */
+export const useSqlQueryRun = () => {
+    const { showToastError } = useToaster();
+    const projectUuid = useAppSelector((state) => state.sqlRunner.projectUuid);
+    const queryClient = useQueryClient();
+    const sqlQueryRunMutation = useMutation<
+        ApiJobScheduledResponse['results'],
+        ApiError,
+        {
+            projectUuid: string;
+            sql: SqlRunnerBody['sql'];
+        }
+    >(({ sql }) => scheduleSqlJob({ projectUuid, sql }), {
+        mutationKey: ['sqlRunner', 'run'],
+    });
+
+    const { data: sqlQueryJob } = sqlQueryRunMutation;
+
+    const { data: scheduledDeliveryJobStatus } = useQuery<
+        ApiJobStatusResponse['results'] | undefined,
+        ApiError
+    >(
+        ['jobStatus', sqlQueryJob?.jobId],
+        () => {
+            if (!sqlQueryJob?.jobId) return;
+            return getSchedulerJobStatus(sqlQueryJob.jobId);
+        },
+        {
+            refetchInterval: (data) => {
+                if (
+                    data?.status === SchedulerJobStatus.COMPLETED ||
+                    data?.status === SchedulerJobStatus.ERROR
+                )
+                    return false;
+                return 2000;
+            },
+            onSuccess: (data) => {
+                if (data?.status === SchedulerJobStatus.ERROR) {
+                    showToastError({
+                        title: 'Could not run SQL query',
+                        subtitle: data.details?.error,
+                    });
+                }
+            },
+            onError: async () => {
+                await queryClient.cancelQueries([
+                    'jobStatus',
+                    sqlQueryJob?.jobId,
+                ]);
```

Comment:
Curious, why does it need to cancel itself?

Reply:
Good point, I removed this and instead stop the `refetch`ing if there's an error on the `query`: https://github.com/lightdash/lightdash/pull/10608/commits/a670d2b3b647ae1f3acc3c19e4fa91f1c429ad47 lmk what you think!